public/components/model_list/index.tsx

```diff
@@ -65,49 +71,81 @@ export const ModelList = ({ notifications }: { notifications: CoreStart['notific
       if (
-        currentPage === previousValue.currentPage &&
-        pageSize === previousValue.pageSize &&
+        newPagination?.currentPage === previousValue.currentPage &&
+        newPagination?.pageSize === previousValue.pageSize &&
         (!sort || sort === previousValue.sort)
       ) {
         return previousValue;
       }
       return {
         ...previousValue,
-        currentPage,
-        pageSize,
+        ...(newPagination
+          ? { currentPage: newPagination.currentPage, pageSize: newPagination.pageSize }
+          : {}),
         ...(sort ? { sort } : {}),
       };
     });
   }, []);
 
+  const handleReset = useCallback(() => {
+    setParams((prevParams) => ({
+      ...prevParams,
+      filterValue: { tag: [], owner: [], stage: [] },
+    }));
+    if (searchInputRef.current) {
+      searchInputRef.current.value = '';
+    }
+  }, [setParams]);
+
   const handleFilterChange = useCallback((filterValue: ModelListFilterFilterValue) => {
     setParams((prevValue) => ({ ...prevValue, filterValue, currentPage: 1 }));
   }, []);
+
   return (
     <EuiPanel>
-      <EuiPageHeader pageTitle={<>Models</>} rightSideItems={[<RegisterNewModelButton />]} />
-      <EuiSpacer />
-      <ModelListFilter value={params.filterValue} onChange={handleFilterChange} />
-      <EuiSpacer />
-      <UploadCallout models={['image-classifier']} />
-      <EuiSpacer />
-      <ModelTable
-        sort={params.sort}
-        models={models}
-        pagination={pagination}
-        onChange={handleTableChange}
-        onModelNameClick={handleViewModelDrawer}
+      <EuiPageHeader
+        pageTitle={
+          <>
+            Models&nbsp;
+            {totalModelCounts && (
+              <EuiTextColor data-test-subj="modelTotalCount" color="subdued" component="span">
+                ({totalModelCounts})
+              </EuiTextColor>
+            )}
+          </>
+        }
+        description="Discover, manage, and track machine learning models across your organization."
+        rightSideItems={[<RegisterNewModelButton />]}
       />
-      <ModelConfirmDeleteModal ref={confirmModelDeleteRef} onDeleted={handleModelDeleted} />
-      {drawerModelName && (
-        <ModelDrawer onClose={() => setDrawerModelName('')} name={drawerModelName} />
+      <EuiSpacer />
+      {!showEmptyScreen && (
+        <>
+          <ModelListFilter
+            value={params.filterValue}
+            onChange={handleFilterChange}
+            searchInputRef={setSearchInputRef}
+          />
+          <EuiSpacer />
+          <ModelTable
+            loading={loading}
+            sort={params.sort}
+            models={models}
+            pagination={pagination}
+            onChange={handleTableChange}
+            onModelNameClick={handleViewModelDrawer}
+            onResetClick={handleReset}
+            errored={!!error}
+          />
+          <ModelConfirmDeleteModal ref={confirmModelDeleteRef} onDeleted={handleModelDeleted} />
```

Comment:
This component is out of date. It will be removed in the official release. The main purpose to call `show` on `ref` is to avoid the component re-render in `ModelList` level. We can consider separate a basic UI modal for all the confirmation usage in model registry.